sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaPruning.scala

```diff
@@ -196,6 +196,7 @@ private[sql] object ParquetSchemaPruning extends Rule[LogicalPlan] {
    */
   private def getRootFields(expr: Expression): Seq[RootField] = {
     expr match {
+      case IsNotNull(_: Attribute) | IsNull(_: Attribute) => Seq.empty
```

Comment:
Hmmmm .. shouldn't we exclude this for filters only in `identifyRootFields`?

Reply:
If this is in projects, I think we also don't need to include all nested fields?

Reply:
But the case mentioned here looks specific to the pushed filter itself. Can we add a simple test for project case as well?

Reply:
I mean, for instance, this case `select address is not null, name.last from contacts` it wouldn't work. I thought this is a quick bandaid fix to resolve a basic case.

Reply:
Thanks. This was a case I didn't test. Fixed it and added test case.